### PR TITLE
Remove the workspace's collection foreign key.

### LINKED ISSUE #69: [production] ActiveRecord::InvalidForeignKey on collection deletion
When deleting a collection, we get a 500 server error. We may need to use the `has_many ... dependent: ...` call in the collections model.

See below

[mirador_sul/production] ActiveRecord::InvalidForeignKey: PG::ForeignKeyViolation: ERROR:  update or delete on table "collections" violates foreign key constraint "fk_rails_0f16b7366a" on table "workspaces"DETAIL:  Key (id)=(1) is still referenced from table "workspaces".: DELETE FROM "collections" WHERE "collections"."id" = $1

## Backtrace

    line 49 of [PROJECT_ROOT]/app/controllers/collections_controller.rb: destroy

[View full backtrace and more info at honeybadger.io](https://app.honeybadger.io/projects/51654/faults/33222324)

### PR DESCRIPTION
**Pull request by jkeck**
Fixes #69

### CODE REVIEW

File: db/migrate/20170410164012_remove_workspaces_collection_foreign_key.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class RemoveWorkspacesCollectionForeignKey < ActiveRecord::Migration[5.0]
+  def change
+    remove_foreign_key :workspaces, :collection
```
Comment (jkeck) [inline db/migrate/20170410164012_remove_workspaces_collection_foreign_key.rb]: The migration doesn't complain, but I don't think we'll know for sure until we try to run this migration under a database that supports foreign key constraints.

Comment (drh-stanford) [inline db/migrate/20170410164012_remove_workspaces_collection_foreign_key.rb]: i think we also need to add `:dependent => :nullify` for the model's relation, right?

Comment (drh-stanford) [inline db/migrate/20170410164012_remove_workspaces_collection_foreign_key.rb]: i think we need to change the model's relation to `has_many ... dependent: :nullify`, right?  See http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many

Comment (drh-stanford) [inline db/migrate/20170410164012_remove_workspaces_collection_foreign_key.rb]: i dug around a bit and i think we can keep the foreign key by changing its `on_delete` behavior with `:nullify` like so:

```
remove_foreign_key :workspaces, :collection
add_foreign_key :workspaces, :collection, on_delete: :nullify
```

As I understand it, we wouldn't need to change the model code at all.

See http://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/SchemaStatements.html#method-i-add_foreign_key

Comment (jkeck) [inline db/migrate/20170410164012_remove_workspaces_collection_foreign_key.rb]: Neat.  I think I have updated the migration to reflect that.  Have a look.

**Comment by jkeck** [comment]
@drh-stanford updated to `:collections`

**Pull request merged**